### PR TITLE
Add transporters spoiler tab for dread

### LINKED ISSUE #5097: Show transporter tab in spoiler window for transport rando
All info is already in the log, just needs UI added.

### PR DESCRIPTION
**Pull request by ThanatosGit**
Also fixed wrong elevators for echoes iff portal rando is selected
Fixes #5097

### CODE REVIEW

File: randovania/games/common/prime_family/gui/prime_trilogy_teleporter_details_tab.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+class PrimeTrilogyTeleporterDetailsTab(TeleporterDetailsTab):
+    @classmethod
+    def should_appear_for(cls, configuration: BaseConfigurator, all_patches: dict[int, GamePatches],
+                          players: PlayersConfiguration) -> bool:
+        assert isinstance(configuration, PrimeConfiguration | EchoesConfiguration)
+        return not configuration.teleporters.is_vanilla
+
+    def tab_title(self) -> str:
+        if self.game_enum == RandovaniaGame.METROID_PRIME_CORRUPTION:
+            return "Teleporters"
```
[the remark this answers is not in the document]
Comment (ThanatosGit) [reply randovania/games/common/prime_family/gui/prime_trilogy_teleporter_details_tab.py]: I just nuked it. Corruption has not even an option to randomise this stuff.

**Review by henriquegemignani — APPROVED** [review]

File: randovania/games/dread/game_data.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 from __future__ import annotations
 
 from randovania.games import game
+from randovania.games.dread.gui.dread_teleporter_details_tab import DreadTeleporterDetailsTab
```
Comment (henriquegemignani) [inline randovania/games/dread/game_data.py:4]: this import needs to be inside `_gui`

Comment (henriquegemignani) [reply randovania/games/dread/game_data.py:4]: actually just import in `dread/gui/__init__.py` and use `gui.DreadTeleporterDetailsTab`

Comment (ThanatosGit) [reply randovania/games/dread/game_data.py]: Like this?
Can you tell me why it has to be this way? I don't really understand the reason atm.

Comment (duncathan) [reply randovania/games/dread/game_data.py]: we support headless installation and operation with a limited feature set, so anything that depends on qt must only be imported when running in gui mode

Comment (henriquegemignani) [reply randovania/games/dread/game_data.py]: Most importantly, the _server_. It's the test that failed even!

**Pull request merged**